hotsos/core/host_helpers/config.py

```diff
@@ -100,62 +97,61 @@ def bool_str(val):
         return val
 
     @property
-    def all(self):
-        return self._sections
+    def all_keys(self):
+        return [x for option in self.config.items()
+                for x in list(option[1].keys())]
 
     def get(self, key, section=None, expand_to_list=False):
-        """ If section is None use flattened """
-        if section is None:
-            value = self._flattened_config.get(key)
-        else:
-            if section not in self._sections:
-                log.debug("section '%s' not found in config file, "
-                          "trying lower case", section)
-                section = section.lower()
-
-            value = self._sections.get(section, {}).get(key)
-
-        if expand_to_list:
-            return self.expand_value_ranges(value)
-
-        return value
-
-    @property
-    def dump(self):
-        with open(self.path) as fd:
-            return fd.read()
+        """ If section is None, then search all sections and return first
+        match. """
+        log.debug("ini read call -> %s:%s", section, key)
+
+        if not self.config:
+            return None
+
+        def post_processing(v):
+            # Sanitize the string and perform
+            # boolean type conversion if needed
+            if isinstance(v, str):
+                v = v.strip()
+                for char in ["'", '"']:
+                    v = v.strip(char)
+                v = self.bool_str(v)
+
+            if expand_to_list:
+                return self.expand_value_ranges(v)
+
+            return v
+
+        # 1: Section is specified.
+        if section is not None:
+            # Look for section name, case-insensitive
+            for sect in self.config.sections() + ["DEFAULT"]:
```

Comment:
When section is given, do we need DEFAULT here?

Reply:
Since we're not using the `get()` to retrieve data, we need "DEFAULT" as default is a valid section name that the user can explicitly give.